File: 1179.py
par = []; impar = []
for i in range(0, 15):
    n = int(input())
    if n % 2 != 0:
        impar += [n]
        if len(impar) == 5:
            for m in range(0, 5):
                print(f'impar[{m}] = {impar[m]}')
            impar.clear()
    else:
        par += [n]
        if len(par) == 5:
            for m in range(0, 5):
                print(f'par[{m}] = {par[m]}')
            par.clear()
if len(impar) != 0:
    for i in range(0, len(impar)):
        print(f'impar[{i}] = {impar[i]}')
if len(par) != 0:
    for i in range(0, len(par)):
        print(f'par[{i}] = {par[i]}')

File: 1179.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');
function clear(n){
    while(n.length){
        n.pop();
    }
    return n;
}

let par = []; let impar = [];
for(var i = 0; i < 15; i++){
    let n = parseInt(lines.shift());
    if(n % 2 !== 0){
        impar.push(n);
        if(impar.length === 5){
            for(var m = 0; m <  5; m++){
                console.log(`impar[${m}] = ${impar[m]}`);
            }
            clear(impar);
        }
    }
    else{
        par.push(n);
        if(par.length === 5){
            for(var m = 0; m < 5; m++){
                console.log(`par[${m}] = ${par[m]}`);
            }
            clear(par);
        }
    }
}
if(impar.length !== 0){
    for(var i = 0; i < impar.length; i++){
        console.log(`impar[${i}] = ${impar[i]}`);
    }
}
if(par.length !== 0){
    for(var i = 0; i < par.length; i++){
        console.log(`par[${i}] = ${par[i]}`);
    }
}
